### PR TITLE
[WIP] Compound distribution PDF

### PR DESCRIPTION
**Pull request by akash9712**
#### Brief description of what is fixed or changed
PDF of distributions with random variable(s) as their parameters can now
be calculated. This is done by marginalizing over the latent distribution.

### CODE REVIEW

File: sympy/stats/crv_types.py [hunk 1]
```diff
@@ -157,7 +157,9 @@ class ArcsinDistribution(SingleContinuousDistribution):
     _argnames = ('a', 'b')
 
     def pdf(self, x):
-        return 1/(pi*sqrt((x - self.a)*(self.b - x)))
+        _pdf = 1/(pi*sqrt((x - self.a)*(self.b - x)))
+        _pdf = self.handle_compound_rv(_pdf)
```
Comment (Upabjojr) [inline sympy/stats/crv_types.py]: I believe we should not modify the `pdf`

**Comment by Upabjojr** [comment]
```python
In [10]: X = Uniform("X", 2, 4)

In [11]: N3 = Normal("N3", X, 2)

In [12]: N3.pspace
Out[12]: SingleContinuousPSpace(N3, NormalDistribution(X, 2))

In [13]: N3.pspace.distribution
Out[13]: NormalDistribution(X, 2)
```

I don't believe that the distribution should be `NormalDistribution(X, 2)`

**Comment by akash9712** [comment]
@Upabjojr 
> I don't believe that the distribution should be NormalDistribution(X, 2)

What should the distribution be instead? Returning a `ContinuousDistributionHandmade` with a modified PDF is one approach, but the random variable returned would then be independent of the latent distributions, which would lead to test failures in tests such as the follows:
```
mu = Normal('mu', 2, 3)
meas = Normal('T', mu, 1)
assert given(meas, Eq(mu, 5)) == Normal('T', 5, 1)
```

**Comment by Upabjojr** [comment]
> What should the distribution be instead? Returning a ContinuousDistributionHandmade with a modified PDF is one approach, but the random variable returned would then be independent of the latent distributions, which would lead to test failures in tests

I would expect the distribution to be independent of `RandomSymbol` objects. Furthermore, if the distribution is called `NormalDistribution`, it's supposed to be a normal distribution, not something else.

What about using the joint distribution followed by a marginalization?

**Comment by Upabjojr** [comment]
Shall we close in favor of #14888

**Pull request closed without merging**